### DISCUSSION TITLE
How to concatenate embedded chapters when merging MKVs?

### DISCUSSION THREAD

**Discussion by JodanJodan** [mifi/lossless-cut#1812] [Q&A]
I have no issue cutting chapters from and re-exporting an MKV with embedded chapters, but when I try to merge multiple MKVs with embedded chapters, the chapters/segments are deleted and the concatenated file just has one chapter per input file. How do I concatenate the chapters? On export, `Merge options` > `Create chapters from merged segments` just does the above, and unchecking it results in no chapters.

**Comment by mifi** [comment]
Hmm, maybe ffmpeg doesn't support merging chapters. Did you try to check the "Include all tracks" merge option? and "Preserve original metadata when merging (slow)".

**Comment by JodanJodan** [comment]
Yes, no permutation of export options seems to produce the desired results. Might mkvmerge be used for this functionality in LosslessCut if not supported by FFmpeg? MKVToolNix properly retimes/combines chapters when losslessly cutting/appending.

**Reply by mifi** [reply]
not trivial to integrate mkvmerge into losslesscut. and i'm not familiar with that tool
